controllers/configmap_controller_test.go

```diff
@@ -0,0 +1,67 @@
+package controllers
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseHosts(t *testing.T) {
+	r := ConfigMapReconciler{}
+
+	testCases := []struct {
+		name        string
+		input       map[string]string
+		expectedOut []*instanceInfo
+		expectedErr bool
+	}{
+		{
+			name:        "invalid username",
+			input:       map[string]string{"localhost": "notusername=core"},
```

Comment:
What about `map[string]string{"invalid": "invalid"}`

Reply:
Thanks for adding it. I was asking for a separate case but was not sure where to place the comment.